wp-content/civi-extensions/goonjcustom/Civi/InductionService.php

```diff
@@ -69,108 +73,171 @@ public static function volunteerCreated(string $op, string $objectName, int $obj
     \Civi::log()->info('Volunteer set: ', [
       'id' => self::$volunteerId,
     ]);
-
   }
 
   /**
-   *
+   * Common logic for creating an induction.
    */
-  public static function createInductionForVolunteer(string $op, string $objectName, int $objectId, &$objectRef) {
-    if ($op !== 'create' || $objectName !== 'Address') {
+  private static function createInduction(int $contactId, int $stateId) {
+    if (self::inductionExists($contactId)) {
+      \Civi::log()->info('Induction already exists for contact', ['id' => $contactId]);
       return FALSE;
     }
 
-    \Civi::log()->info('Address created: ', [
-      'id' => $objectId,
-      'volunteer' => self::$volunteerId,
-      'contactId' => $objectRef->contact_id,
-      'isPrimary' => $objectRef->is_primary,
+    if (!$stateId) {
+      return FALSE;
+    }
```

Comment:
I don't think you need this. You have type hint on the function `int $stateId` so it will error if the stateId is null or something